src/phoenix/trace/dsl/query.py

```diff
@@ -268,75 +439,129 @@ def rename(self, **kwargs: str) -> "SpanQuery":
 
     def with_index(self, key: str = "context.span_id") -> "SpanQuery":
         _index = Projection(key=key)
-        return replace(self, _index=_index)
+        return replace(self, _index=_index, _explode=replace(self._explode, primary_index_key=key))
 
     def with_concat_separator(self, separator: str = "\n\n") -> "SpanQuery":
         _concat = self._concat.with_separator(separator)
         return replace(self, _concat=_concat)
 
-    def with_explode_primary_index_key(self, primary_index_key: str) -> "SpanQuery":
-        _explode = self._explode.with_primary_index_key(primary_index_key)
-        return replace(self, _explode=_explode)
-
-    def __call__(self, spans: Iterable[Span]) -> pd.DataFrame:
+    def with_explode_primary_index_key(self, _: str) -> "SpanQuery":
+        print(
+            "`.with_explode_primary_index_key(...)` is deprecated and wil be "
+            "removed in the future. Use `.with_index(...)` instead."
+        )
+        return self
+
+    def __call__(
+        self,
+        session: Session,
+        project_name: Optional[str] = None,
+        start_time: Optional[datetime] = None,
+        stop_time: Optional[datetime] = None,
+        root_spans_only: Optional[bool] = None,
+    ) -> pd.DataFrame:
+        if not project_name:
+            project_name = DEFAULT_PROJECT_NAME
+        if not (self._select or self._explode or self._concat):
+            return _get_spans_dataframe(
+                session,
+                self._filter,
+                project_name,
+                start_time,
+                stop_time,
+                root_spans_only,
+            )
+        assert session.bind is not None
+        dialect = session.bind.dialect.name
+        conn = session.connection()
+        index = _NAMES[self._index.key].label(self._add_tmp_suffix(self._index.key))
+        row_id = models.Span.id.label(self._pk_tmp_col_label)
+        stmt = (
+            # We do not allow `group_by` anything other than `row_id` because otherwise
+            # it's too complex for the post hoc processing step in pandas.
+            select(row_id)
+            .join_from(models.Span, models.Trace)
```

Comment:
What is `join_from` doing that is different from `join`?

Reply:
oh this is actually redundant row and i forgot to change it.